algorithms/src/std_algorithms/impl/Kokkos_Unique.hpp

```diff
@@ -105,7 +105,9 @@ IteratorType unique_exespace_impl(const std::string& label,
       // using the same algorithm used for unique_copy but we now move things
       using value_type    = typename IteratorType::value_type;
       using tmp_view_type = Kokkos::View<value_type*, ExecutionSpace>;
-      tmp_view_type tmp_view("std_unique_tmp_view", num_elements_to_explore);
+      tmp_view_type tmp_view(Kokkos::view_alloc(ex, Kokkos::WithoutInitializing,
```

Comment:
Please update the description and say this also skip initialization.

Reply:
Ok, done :wink:

Reply:
why do we want to skip the initialization?

Reply:
@fnrizzi I would say that for "scalar" types (a `double` for instance), it might not make a difference. For a custom type, default constructing it using the `ViewValueFunctor` might be a cost we want to avoid. Also note that when initializing, I think what happens is really backend-dependent and is driven by `ZeroMemset`.

All in all, there might be no noticeable performance improvement, but at least now the view is initialized correctly (I mean the following code does not require that the view is zero initialized).

Reply:
i was for a moment concerned about first touch allocation, but since the tmp_view is used immediately below in the par_scan without doing anything particular, i guess this is fine